src/main/kotlin/org/jitsi/nlj/rtp/codec/vp8/Vp8Packet.kt

```diff
@@ -24,10 +24,36 @@ import org.jitsi.rtp.extensions.bytearray.hashCodeOfSegment
 import org.jitsi_modified.impl.neomedia.codec.video.vp8.DePacketizer
 
 class Vp8Packet(
-    data: ByteArray,
+    buffer: ByteArray,
     offset: Int,
     length: Int
-) : VideoRtpPacket(data, offset, length) {
+) : VideoRtpPacket(buffer, offset, length) {
+    override val isKeyframe: Boolean by lazy(LazyThreadSafetyMode.PUBLICATION) {
+        DePacketizer.isKeyFrame(buffer, payloadOffset, payloadLength)
+    }
+
+    /**
+     * This is currently used as an overall spatial index, not an in-band spatial quality index a la vp9.  That is,
+     * this index will correspond to an overall simulcast layer index across multiple simulcast stream.  e.g.
+     * 180p stream packets will have 0, 360p -> 1, 720p -> 2
+     */
+    val spatialLayerIndex: Int by lazy(LazyThreadSafetyMode.PUBLICATION) {
+        if (isKeyframe) Vp8Utils.getSpatialLayerIndexFromKeyFrame(this) else -1
+    }
+
+    /**
+     * We alias qualityIndex to spatialLayerIndex for VP8.
+     */
+    override val qualityIndex: Int
+        get() = spatialLayerIndex
+
+    val isStartOfFrame: Boolean
```

Comment:
why not lazy for these as well?

Reply:
For isStartOfFrame, we could do that.

I wasn't sure if how much sense it made for vals that were direct aliases of other vals.